amtl/am-hashtable.h

```diff
@@ -319,10 +319,18 @@ class HashTable : private AllocPolicy
     Probulator probulator(hash, capacity_);
 
     Entry *e = &table_[probulator.entry()];
+    Entry *firstRemoved = nullptr;
     for (;;) {
-      if (!e->isLive())
+      if (e->isFree()) {
+        if (firstRemoved)
```

Comment:
This is more confusing than just checking at the return site - the data flow is harder to follow.